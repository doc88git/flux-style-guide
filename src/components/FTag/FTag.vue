<template>
  <div class="FTag" :class="classBgColor">
    <div class="FTag__fieldset">
      <div class="FTag__legend" :class="classBgColor">{{ legend }}</div>
      <div class="FTag__slot-div" :class="classLineColor">
        <slot class="FTag__slot">
          <f-icon class="FTag__icon" :name="icon" :color="iconColor" />
          <div class="FTag__text">{{ text }}</div>
        </slot>
      </div>
    </div>
  </div>
</template>

<script>
import FIcon from "../FIcon/FIcon";

export default {
  name: "FTag",
  components: {
    FIcon
  },
  props: {
    bgColor: String,
    lineColor: String,
    legend: String,
    icon: String,
    iconColor: String,
    text: String
  },
  computed: {
    classBgColor() {
      return this.bgColor ? `FTag--${this.bgColor}` : "";
    },
    classLineColor() {
      return this.lineColor ? `FTag__fieldset--${this.lineColor}` : "";
    }
  }
};
</script>

<style lang="scss" scoped>
.FTag {
  &__fieldset {
    display: flex;
    &:hover {
      .FTag__legend {
        display: block;
      }
    }
  }

  &__legend {
    display: none;
    padding-right: 7px;
    padding-left: 3px;
    padding-bottom: 0;
    position: absolute;
    transform: translateY(-9px);
    margin-left: 0;
    font-size: 12px;
    z-index: 2;
    background: #ffffff;
  }

  &__slot-div {
    align-items: center;
    border: 1px solid;
    padding-top: 4px;
    padding-right: 10px;
    padding-left: 7px;
    padding-bottom: 4px;
    border-radius: 5px;
    display: flex;
    min-width: 65px;
    justify-content: center;
  }

  &__icon {
    padding-top: 0;
    padding-left: 1px;
    padding-right: 5px;
    padding-bottom: 1px;
  }
}

@import "../../assets/f-colors.scss";
</style>
