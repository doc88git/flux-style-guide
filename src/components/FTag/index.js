import FTag from './FTag'
import FTagGroup from './FTagGroup'

export { FTag, FTagGroup }
