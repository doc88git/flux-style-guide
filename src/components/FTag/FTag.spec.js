import { shallowMount } from '@vue/test-utils'
import FTag from './FTag'

const WRAPPER_PROPS = {
  bgColor: 'white',
  lineColor: '#C1C1C1',
  legend: 'Empresa',
  icon: 'dashboard',
  iconColor: 'purple',
  text: 'Flux Services',
  textColor: '#7F7F7F'
}

describe('FTag tests', () => {
  let WRAPPER

  beforeEach(() => (WRAPPER = shallowMount(FTag, { propsData: WRAPPER_PROPS })))

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Componente não alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })

  describe('Teste de computeds', () => {
    describe('Teste da computed styleLegend', () => {
      test('styleLegend durante evento mouseOut', () => {
        const fTagLegend = WRAPPER.find('.FTag__legend')
        const styleExpect = 'background-color: white; visibility: hidden;'

        expect(fTagLegend.attributes('style')).toEqual(styleExpect)
      })

      test('styleLegend durante evento mouseOver', async () => {
        const fTagFieldset = WRAPPER.find('.FTag__fieldset')
        const fTagLegend = WRAPPER.find('.FTag__legend')
        const styleExpect = 'background-color: white; visibility: visible;'

        fTagFieldset.trigger('mouseover')
        await WRAPPER.vm.$nextTick()

        expect(fTagLegend.attributes('style')).toEqual(styleExpect)
      })
    })

    test('Teste da computed styleLineColor', () => {
      const computedExpect = { borderColor: '#C1C1C1', minWidth: '10px' }
      expect(WRAPPER.vm.styleLineColor).toEqual(computedExpect)
    })

    test('Teste da computed styleTextColor', () => {
      expect(WRAPPER.vm.styleTextColor).toBe('color: #7F7F7F')
    })

    test('Teste da computed styleBeforeColor', () => {
      expect(WRAPPER.vm.styleBeforeColor).toBe(':before: #C1C1C1')
    })
  })

  describe('Teste de métodos', () => {
    test('Teste do método setLegendSize', () => {
      WRAPPER.vm.setLegendSize()
      expect(typeof WRAPPER.vm.legendSize).toBe('number')
    })
  })
})
