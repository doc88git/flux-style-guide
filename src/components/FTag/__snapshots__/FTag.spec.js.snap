// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FTag tests Componente n√£o alterado 1`] = `
"<div class=\\"FTag\\">
  <div class=\\"FTag__fieldset\\" style=\\"color: rgb(127, 127, 127);\\">
    <div class=\\"FTag__legend\\" style=\\"background-color: white; visibility: hidden;\\">
      Empresa
    </div>
    <div class=\\"FTag__slot-div\\" style=\\"border-color: #c1c1c1; min-width: 10px;\\">
      <f-icon-stub color=\\"purple\\" lib=\\"material\\" size=\\"base\\" name=\\"dashboard\\" class=\\"FTag__icon\\"></f-icon-stub>
      <div class=\\"FTag__text\\">
        Flux Services
      </div>
    </div>
  </div>
</div>"
`;
